presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergNativeMetadata.java

```diff
@@ -0,0 +1,536 @@
+        transaction.commitTransaction();
+
+        return Optional.of(new IcebergWrittenPartitions(commitTasks.stream()
+                .map(CommitTaskData::getPath)
+                .collect(toImmutableList())));
+    }
+
+    @Override
+    public ColumnHandle getUpdateRowIdColumnHandle(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        return new IcebergColumnHandle(0, "$row_id", BIGINT, Optional.empty());
+    }
+
+    @Override
+    public void dropTable(ConnectorSession session, ConnectorTableHandle tableHandle)
+    {
+        TableIdentifier tableIdentifier = toIcebergTableIdentifier(((IcebergTableHandle) tableHandle).getSchemaTableName());
+        resourceFactory.getCatalog(session).dropTable(tableIdentifier);
+    }
+
+    @Override
+    public void renameTable(ConnectorSession session, ConnectorTableHandle tableHandle, SchemaTableName newTable)
+    {
+        TableIdentifier from = toIcebergTableIdentifier(((IcebergTableHandle) tableHandle).getSchemaTableName());
+        TableIdentifier to = toIcebergTableIdentifier(newTable);
+        resourceFactory.getCatalog(session).renameTable(from, to);
+    }
+
+    @Override
+    public void addColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnMetadata column)
+    {
+        TableIdentifier tableIdentifier = toIcebergTableIdentifier(((IcebergTableHandle) tableHandle).getSchemaTableName());
+        Table icebergTable = resourceFactory.getCatalog(session).loadTable(tableIdentifier);
+        icebergTable.updateSchema().addColumn(column.getName(), toIcebergType(column.getType())).commit();
+    }
+
+    @Override
+    public void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle column)
+    {
+        TableIdentifier tableIdentifier = toIcebergTableIdentifier(((IcebergTableHandle) tableHandle).getSchemaTableName());
+        Table icebergTable = resourceFactory.getCatalog(session).loadTable(tableIdentifier);
+        IcebergColumnHandle handle = (IcebergColumnHandle) column;
+        icebergTable.updateSchema().deleteColumn(handle.getName()).commit();
+    }
+
+    @Override
+    public void renameColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle source, String target)
+    {
+        TableIdentifier tableIdentifier = toIcebergTableIdentifier(((IcebergTableHandle) tableHandle).getSchemaTableName());
+        Table icebergTable = resourceFactory.getCatalog(session).loadTable(tableIdentifier);
+        IcebergColumnHandle columnHandle = (IcebergColumnHandle) source;
+        icebergTable.updateSchema().renameColumn(columnHandle.getName(), target).commit();
+    }
+
+    private ConnectorTableMetadata getTableMetadata(ConnectorSession session, SchemaTableName table)
+    {
+        Table icebergTable;
+        try {
+            icebergTable = resourceFactory.getCatalog(session).loadTable(toIcebergTableIdentifier(table));
+        }
+        catch (NoSuchTableException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)